producers/http/handlers.go

```diff
@@ -89,9 +89,12 @@ func containerHandler(p *producerImpl) http.HandlerFunc {
 			return
 		}
 
-		httpLog.Debugf("Encoding container metrics:\n%+v", containerMetrics)
-
-		encode(containerMetrics, w)
+		combinedMetrics, err := combineMessages(containerMetrics)
+		if err != nil {
+			httpLog.Errorf("/v0/containers/{id} - %s", err.Error())
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
```

Comment:
Elsewhere after checking for `err` you also do a check on `len(combinedMetrics)` and if `0` return an error.  Was leaving that part out here intentional?